components/event-publisher-proxy/pkg/subscribed/helpers.go

```diff
@@ -72,10 +91,49 @@ func AddUniqueEventsToResult(eventsSubSet []Event, uniqEvents map[Event]bool) ma
 }
 
 // FilterEventTypeVersions returns a slice of Events:
+// 1. if the eventType matches the format for standard type matching: <event-name>.<version>
+// E.g. order.created.v0
+// 2. if the eventType matches the format for exact type matching: <eventTypePrefix><appName>.<event-name>.<version>
+// E.g. sap.external.custom.exampleapp.order.created.v0
+func FilterEventTypeVersions(appName string, subscription *eventingv1alpha2.Subscription) []Event {
+	events := make([]Event, 0)
+	for _, eventType := range subscription.Spec.Types {
+		if subscription.Spec.TypeMatching == eventingv1alpha2.TypeMatchingExact {
+			// in case of type matching exact, we have app name as a part of event type
+			appNameWithDots := "." + appName + "."
+			if strings.Contains(eventType, appNameWithDots) {
+				eventTypeVersionArr := strings.Split(eventType, appNameWithDots)
+				eventTypeVersion := eventTypeVersionArr[len(eventTypeVersionArr)-1]
```

Comment:
Maybe optimize these two lines into one as:
```
eventTypeVersion  := strings.SplitAfter(eventType, appNameWithDots)[1]
```

Reply:
I haven't tested this. Just a hint! Please try it out and see if there are edge cases

Reply:
I think it wouldn't work if the appName is `kyma` and the prefix contains `kyma`. Maybe a testcase to add :)